src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -310,19 +312,14 @@ private static IntPtr GetSslMethod(bool isServer, long options)
             return method;
         }
 
-        //TODO See if SSL_CTX_Set_quite_shutdown can be used
         private static void Disconnect(IntPtr sslPtr)
         {
             if (IntPtr.Zero != sslPtr)
             {
                 int retVal = libssl.SSL_shutdown(sslPtr);
-                if (0 == retVal)
+                if (retVal < 0)
                 {
-                    retVal = libssl.SSL_shutdown(sslPtr);
-                    if (retVal < 0)
-                    {
-                        libssl.SSL_get_error(sslPtr, retVal);
-                    }
+                    libssl.SSL_get_error(sslPtr, retVal);
```

Comment:
Since error value is not used, the if can be converted to a Debug.Assert(retVal == 1, "SSL shutdown failed with " + libssl.SSL_get_error(sslPtr, retVal))

LGTM otherwise